jwst/resample/resample_step.py

```diff
@@ -128,10 +136,11 @@ def process(self, input):
 
     def update_phot_keywords(self, model):
         """Update pixel scale keywords"""
+        print('inside', self.pixel_scale_ratio)
         if model.meta.photometry.pixelarea_steradians is not None:
-            model.meta.photometry.pixelarea_steradians *= self.pixel_scale_ratio**2
+            model.meta.photometry.pixelarea_steradians *= model.meta.resample.pixel_scale_ratio**2
```

Comment:
The fact that you're taking the square of `pixel_scale_ratio` here suggests that it's in linear units, as opposed to the ratio of the pixel areas. But up above it looked like it was being handled as if it were already a ratio of areas. Check for consistency. Maybe I'm just losing track of the units during the step flow.

Reply:
Based on more recent info it appears that `pixel_scale_ratio` is in fact in linear space, not in terms of area, so the only remaining question is whether this scaling is in the right sense. Given that the documentation says that a `pixel_scale_ratio` value of 0.5 results in 4 output pixels for every 1 input pixel, the output pixel area is in fact 4 times smaller and hence multiplying the original area by 0.5*0.5=0.25 appears to be the correct thing to do.